hibernate-core/src/main/java/org/hibernate/sql/results/internal/StandardEntityGraphTraversalStateImpl.java

```diff
@@ -96,6 +98,25 @@ else if ( graphSemantic == GraphSemantic.FETCH ) {
 		return new TraversalResult( previousContextRoot, fetchStrategy );
 	}
 
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	private GraphImplementor<?> getMergedSubGraphs(
+			Map<? extends Class<?>, ? extends SubGraphImplementor<?>> subgraphMap,
+			Class<?> subgraphMapKey) {
+		ManagedDomainType<?> managedDomainType = metamodel.findManagedType( subgraphMapKey );
+
+		if ( managedDomainType == null ) {
+			return null;
+		}
+
+		GraphImplementor mergedSubgraph = new SubGraphImpl<>( managedDomainType, true );
+
+		for ( GraphImplementor subgraph : subgraphMap.values() ) {
+			mergedSubgraph.merge( subgraph );
```

Comment:
This will not be necessary anymore once the `SubGraphImpl` knows of its parent like the spec requires it. See my previous comment.

Reply:
yes, you're absolutely right